internal/compare/comparation.go

```diff
@@ -34,3 +36,13 @@ func Contains[T comparable](a []T, b T) bool {
 	}
 	return false
 }
+
+func ContainsDeepEqual[T comparable](a []T, b T) bool {
```

Comment:
this is not necessary, `reflect.DeepEquals` does what you are doing on slices.

Reply:
I simple `reflect.DeepEquals` should be enough, see https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1929/files#diff-11a090be55936627141c551ed7808292f8dc76db3baece47ad78ad24fb89d730R28